src/UniversalFunctions.jl

```diff
@@ -491,17 +537,209 @@ function psi(uf::Grachev, ζ, tt::HeatTransport)
+# default. 
+
+# Fields
+
+$(DSE.FIELDS)
+"""
+struct Cheng{FT, PS <: ChengParams} <: AbstractUniversalFunction{FT}
+    "Monin-Obhukov Length"
+    L::FT
+    params::PS
+end
+
+struct ChengType <: AbstractUniversalFunctionType end
+Cheng() = ChengType()
+
+# Nishizawa2018 Eq. A7
+f_momentum(uf::Cheng, ζ) = sqrt(sqrt(1 - 15 * ζ))
+
+# Nishizawa2018 Eq. A8
+f_heat(uf::Cheng, ζ) = sqrt(1 - 9 * ζ)
+
+function phi(uf::Cheng, ζ, tt::MomentumTransport)
+    FT = eltype(uf)
+    if ζ > 0
+        # Cheng2005 Eq. 22
+        FT = eltype(uf)
+        _a_m = FT(a_m(uf))
+        _b_m = FT(b_m(uf))
+        num = ζ + (ζ^_b_m) * (1 + ζ^_b_m)^((1 - _b_m) / _b_m)
+        denom = ζ + (1 + ζ^_b_m)^(1 / _b_m)
+        return FT(1) + _a_m * (num / denom)
+    else
+        # Nishizawa2018 Eq. A1 (L < 0)
+        f_m = f_momentum(uf, ζ)
+        return 1 / f_m
+    end
+end
+
+function phi(uf::Cheng, ζ, tt::HeatTransport)
+    FT = eltype(uf)
+    if ζ > 0
+        # Cheng2005 Eq. 24
+        FT = eltype(uf)
+        _a_h = FT(a_h(uf))
+        _b_h = FT(b_h(uf))
+        num = ζ + (ζ^_b_h) * (1 + ζ^_b_h)^((1 - _b_h) / _b_h)
+        denom = ζ + (1 + ζ^_b_h)^(1 / _b_h)
+        return FT(1) + _a_h * (num / denom)
+
+    else
+        # Nishizawa2018 Eq. A2 (L < 0)
+        f_h = f_heat(uf, ζ)
+        return 1 / f_h
+    end
+end
+
+function psi(uf::Cheng, ζ, tt::MomentumTransport)
+    FT = eltype(uf)
+    if ζ > 0
+        # Cheng2005 Eq. 21
+        _a_m = FT(a_m(uf))
+        _b_m = FT(b_m(uf))
+        return -_a_m * log(ζ + (1 + ζ^_b_m)^(1 / _b_m))
+    else
+        # Nishizawa2018 Eq. A3 (ζ < 0)
+        f_m = f_momentum(uf, ζ)
+        log_term = log((1 + f_m)^2 * (1 + f_m^2) / 8)
+        return log_term - 2 * atan(f_m) + FT(π) / 2
+
+    end
+end
+
+function psi(uf::Cheng, ζ, tt::HeatTransport)
+    FT = eltype(uf)
+    if ζ > 0
+        # Cheng2005 Eq. 23
+        _a_h = FT(a_h(uf))
+        _b_h = FT(b_h(uf))
+        return -_a_h * log(ζ + (1 + ζ^_b_h)^(1 / _b_h))
+    else
+        # Nishizawa2018 Eq. A4 (ζ < 0)
+        f_h = f_heat(uf, ζ)
+        return 2 * log((1 + f_h) / 2)
+    end
+end
+
+function Psi(uf::Cheng, ζ, tt::HeatTransport)
+    FT = eltype(uf)
+    if abs(ζ) < eps(typeof(uf.L))
+        if ζ >= 0
+            # Volume-averaged form of Cheng2005 Eq. 23
+            # approximated with Nishizawa2018 Eq. B7
+            _a_h = FT(a_h(uf))
+            _b_h = FT(b_h(uf))
+
+            limit_term = _a_h
+            return -(ζ / 2) * limit_term
+        else
+            # Nishizawa2018 Eq. A14 (ζ < 0)
+            return -9 * ζ / 4
+        end
+    else
+        if ζ >= 0
+            # Volume-averaged form of Cheng2005 Eq. 23
+            _a_h = FT(a_h(uf))
+            _b_h = FT(b_h(uf))
+
+            #TODO: Implement volume-averaged form of Cheng2005 Eq. 23 for ζ >> 0
+            @error(
+                "Volume-averaged form of Cheng (2005) heat SCF for ζ >> 0 not yet implemented. Defaulting to Nishizawa (2018) formulation.",
```

Comment:
"Defaulting to Nishizawa..." sounds like this is a warning after which it defaults. Maybe say something like "Use Nishizawa... instead"

Reply:
Is the intent to warn the user ? The error macro is a break point (OK if this is intended - I would then change the error message)- the code will not progress past this statement if it is triggered.

Reply:
Would it be preferred to ```@warn``` and automatically proceed with the Nishizawa (2018) formulation, or ```@error``` and force the user to use a different UF formulation?

Reply:
error and break is fine -> This leaves little ambiguity in the code w.r.t. expected outcomes. `FDScheme()` is supported in the above code throug the lower case `phi, psi` functions.